core/services/job/models.go

```diff
@@ -115,47 +115,51 @@ var (
-	GatewaySpecID                 *int32
+	ID                            int32                        `toml:"-"`
+	ExternalJobID                 uuid.UUID                    `toml:"externalJobID"`
+	BlockhashStoreSpec            *BlockhashStoreSpec          `db:"-"`
+	BlockhashStoreSpecID          *int32                       `db:"-"`
+	BlockHeaderFeederSpec         *BlockHeaderFeederSpec       `db:"-"`
+	BlockHeaderFeederSpecID       *int32                       `db:"-"`
+	BootstrapSpec                 *BootstrapSpec               `db:"-"`
+	BootstrapSpecID               *int32                       `db:"-"`
+	CronSpec                      *CronSpec                    `db:"-"`
+	CronSpecID                    *int32                       `db:"-"`
+	DirectRequestSpec             *DirectRequestSpec           `db:"-"`
+	DirectRequestSpecID           *int32                       `db:"-"`
+	FluxMonitorSpec               *FluxMonitorSpec             `db:"-"`
+	FluxMonitorSpecID             *int32                       `db:"-"`
+	KeeperSpec                    *KeeperSpec                  `db:"-"`
+	KeeperSpecID                  *int32                       `db:"-"`
+	LegacyGasStationServerSpec    *LegacyGasStationServerSpec  `db:"-"`
+	LegacyGasStationServerSpecID  *int32                       `db:"-"`
+	LegacyGasStationSidecarSpec   *LegacyGasStationSidecarSpec `db:"-"`
+	LegacyGasStationSidecarSpecID *int32                       `db:"-"`
+	OCR2OracleSpec                *OCR2OracleSpec              `db:"-"`
+	OCR2OracleSpecID              *int32                       `db:"-"`
+	OCROracleSpec                 *OCROracleSpec               `db:"-"`
+	OCROracleSpecID               *int32                       `db:"-"`
+	VRFSpec                       *VRFSpec                     `db:"-"`
+	VRFSpecID                     *int32                       `db:"-"`
+	WebhookSpec                   *WebhookSpec                 `db:"-"`
+	WebhookSpecID                 *int32                       `db:"-"`
+	GatewaySpec                   *GatewaySpec                 `db:"-"`
+	GatewaySpecID                 *int32                       `db:"-"`
 	PipelineSpecID                int32
 	PipelineSpec                  *pipeline.Spec
 	JobSpecErrors                 []SpecError
 	Type                          Type
-	SchemaVersion                 uint32
-	GasLimit                      clnull.Uint32 `toml:"gasLimit"`
-	ForwardingAllowed             bool          `toml:"forwardingAllowed"`
-	Name                          null.String
-	MaxTaskDuration               models.Interval
-	Pipeline                      pipeline.Pipeline `toml:"observationSource"`
-	CreatedAt                     time.Time
+	// Type spec contains job type specific configuration that is customisable per job.
+	// Each job of one type can configure the same properties, but with different values.
+	// TODO: Document where properties for the type are defined.
+	TypeSpec          JSONConfig
```

Comment:
I'm thinking that we should try to make `interface{}` work here if possible, since it would require the least extra boilerplate:
```suggestion
	TypeSpec          interface{}
```
But if that is not possible (we can't always know the type beforehand?), then we could query as the raw bytes, and just defer unmarshalling until later, when the type is known:
```suggestion
	TypeSpec          []byte
```